ksqldb-engine/src/main/java/io/confluent/ksql/util/ValidationSharedKafkaStreamsRuntimeImpl.java

```diff
@@ -15,192 +15,103 @@
+    );
   }
 
-  public void markSources(final QueryId queryId, final Set<SourceName> sourceNames) {
-    sources.put(queryId, sourceNames);
+  private static Map<String, Object> getSandboxStreamsProperties(
+      final SharedKafkaStreamsRuntime sharedKafkaStreamsRuntime
+  ) {
+    final Map<String, Object> sandboxStreamsProperties =
+        new ConcurrentHashMap<>(sharedKafkaStreamsRuntime.getStreamProperties());
+    sandboxStreamsProperties.put(
+        StreamsConfig.APPLICATION_ID_CONFIG,
+        "_confluent-ksql-" + UUID.randomUUID() + "-validation"
+    );
+    return sandboxStreamsProperties;
   }
 
+  @Override
   public void register(
       final QueryErrorClassifier errorClassifier,
-      final Map<String, Object> streamsProperties,
-      final PersistentQueriesInSharedRuntimesImpl persistentQueriesInSharedRuntimesImpl,
-      final QueryId queryId) {
+      final BinPackedPersistentQueryMetadata binpackedPersistentQueryMetadata,
+      final QueryId queryId
+  ) {
     if (!sources.containsKey(queryId)) {
       if (sources
           .values()
           .stream()
           .flatMap(Collection::stream)
-          .anyMatch(t -> persistentQueriesInSharedRuntimesImpl.getSourceNames().contains(t))) {
+          .anyMatch(t -> binpackedPersistentQueryMetadata.getSourceNames().contains(t))) {
         throw new IllegalArgumentException(
             queryId.toString() + ": was not reserved on this runtime");
       } else {
-        sources.put(queryId, persistentQueriesInSharedRuntimesImpl.getSourceNames());
+        sources.put(queryId, binpackedPersistentQueryMetadata.getSourceNames());
       }
     }
-
-    metadata.put(queryId.toString(), persistentQueriesInSharedRuntimesImpl);
-    LOG.debug("mapping {}", metadata);
+    collocatedQueries.put(queryId, binpackedPersistentQueryMetadata);
+    log.debug("mapping {}", collocatedQueries);
   }
 
+  @Override
   public StreamsUncaughtExceptionHandler.StreamThreadExceptionResponse uncaughtHandler(
       final Throwable e
   ) {
-    return StreamsUncaughtExceptionHandler.StreamThreadExceptionResponse.REPLACE_THREAD;
-  }
-
-  @SuppressFBWarnings(value = "EI_EXPOSE_REP", justification = "must expose streams")
-  public KafkaStreams getKafkaStreams() {
-    return kafkaStreams;
-  }
-
-  public KafkaStreams.State state() {
-    return kafkaStreams.state();
-  }
-
-  public Collection<StreamsMetadata> allMetadata() {
-    return kafkaStreams.metadataForAllStreamsClients();
-  }
-
-  public Set<StreamsTaskMetadata> getTaskMetadata() {
-    return kafkaStreams.metadataForLocalThreads()
-        .stream()
-        .flatMap(t -> t.activeTasks().stream())
-        .map(StreamsTaskMetadata::fromStreamsTaskMetadata)
-        .collect(Collectors.toSet());
-  }
-
-  public boolean isError(final QueryId queryId) {
-    return false;
+    return StreamThreadExceptionResponse.REPLACE_THREAD;
   }
 
+  @Override
   public void stop(final QueryId queryId) {
-
+    log.trace("");
```

Comment:
Maybe fill this out?, I don't think this should be called

Reply:
whoops, thanks for the catch